params/config.go

```diff
@@ -358,24 +386,42 @@ type ChainConfig struct {
 	Ethash *EthashConfig `json:"ethash,omitempty"`
 	Clique *CliqueConfig `json:"clique,omitempty"`
 
-	// Scroll genesis extension: Use zktrie
-	Zktrie bool `json:"zktrie,omitempty"`
+	// Scroll genesis extension: enable scroll rollup-related traces & state transition
+	Scroll ScrollConfig `json:"scroll,omitempty"`
+}
+
+type ScrollConfig struct {
+	// Use zktrie [optional]
+	UseZktrie bool `json:"useZktrie,omitempty"`
 
-	// Scroll genesis extension: Transaction fee vault address [optional]
+	// Maximum number of transactions per block [optional]
+	MaxTxPerBlock *int `json:"maxTxPerBlock,omitempty"`
+
+	// Transaction fee vault address [optional]
 	FeeVaultAddress *common.Address `json:"feeVaultAddress,omitempty"`
 
-	// Scroll genesis extension: enable EIP-2718 in tx pool.
+	// Enable EIP-2718 in tx pool [optional]
 	EnableEIP2718 bool `json:"enableEIP2718,omitempty"`
 
-	// Scroll genesis extension: enable EIP-1559 in tx pool, EnableEIP2718 should be true too.
+	// Enable EIP-1559 in tx pool, EnableEIP2718 should be true too [optional]
 	EnableEIP1559 bool `json:"enableEIP1559,omitempty"`
+}
 
-	// Scroll genesis extension: Maximum number of transactions per block [optional]
-	MaxTxPerBlock *int `json:"maxTxPerBlock,omitempty"`
+func (s ScrollConfig) BaseFeeEnabled() bool {
+	return s.EnableEIP2718 && s.EnableEIP1559
+}
 
-	// Scroll genesis extension: enable scroll rollup-related traces & state transition
-	// TODO: merge with these config: Zktrie, FeeVaultAddress, EnableEIP2718, EnableEIP1559 & MaxTxPerBlock
-	UsingScroll bool `json:"usingScroll,omitempty"`
+func (s ScrollConfig) L1FeeEnabled() bool {
```

Comment:
The function name is a bit inaccurate. Using fee vault & enable L1 fee is different. Fee vault collects the tx fee that includes both L2 exec fee and L1 rollup fee, right?